cartridges/int_algolia_sfra/cartridge/static/default/js/algolia/autocomplete-config.js

```diff
@@ -53,6 +77,65 @@ function enableAutocomplete(config) {
     });
 }
 
+/**
+ * Maps Algolia hit to a trending item object.
+ * @param {Object} hit - The Algolia hit object.
+ * @returns {Object} - The mapped trending item object.
+ */
+function mapHitToTrendingItem(hit) {
+    return {
+        label: hit.name,
+        objectID: hit.objectID,
+        disBaseLink: hit.image_groups[0].images[0].dis_base_link,
+        url: hit.url,
+    };
+}
+
+/**
+ * Fetches and processes trending items.
+ * @param {Object} RecommendConfig - The configuration object.
+ * @returns {Promise} - The promise object.
+ */
+function fetchTrendingItems(RecommendConfig) {
+    return new Promise((resolve, reject) => {
+        const indexName = RecommendConfig.productsIndex;
+
+        recommendClient.getTrendingItems([{
+            indexName,
+            maxRecommendations,
+        }]).then(response => {
+            const results = response.results[0].hits;
+            trendingItemsArr = results.map(mapHitToTrendingItem);
+            resolve(trendingItemsArr);
+        }).catch(err => {
+            console.error('Failed to fetch trending items:', err);
+            reject(err);
+        });
+    });
+}
+
+
+/**
+ * @description Get Trending items array for autocomplete
+ * @param {Object} config configuration object
+ * @returns {Promise} Promise that resolves with the trending items array
+ */
+function getTrendingItemsArray() {
+    if (!algoliaData.enableRecommend) {
+        return Promise.resolve([]);
+    }
+
+    if (trendingItemsArr.length === 0) {
+        const RecommendConfig = {
+            productsIndex: algoliaData.productsIndex,
+            recommendClient: recommendClient,
+        };
+        return fetchTrendingItems(RecommendConfig);
```

Comment:
This doesn't bring much, you don't use the passed `recommendClient` and use the global client instead. Let's simplify:
```suggestion
        return fetchTrendingItems();
```